APE22.rst

```diff
@@ -0,0 +1,440 @@
+Astropy's API Policy on Modules and their Contents
+---------------------------
+
+author: Nathaniel Starkman
+
+date-created: 2013 November 5 <replace with the date you submit the APE>
+
+date-last-revised: 2013 December 17 <keep this up to date anytime something
+changes>
+
+date-accepted: 2014 January 20 <replace with accepted date>
+
+type: <one of these three: Standard Track, Informational, Process>
+
+status: Discussion
+
+
+Abstract
+--------
+
+This APE establishes Astropy's policy for public versus internal modules and
+their contents.
+
+
+Scope
+-----
+
+A library's API is the publicly accessible "surface": the set of modules,
+classes, functions, and other symbols provided to the library's users. This APE
+is narrowly focused on the module level of the API -- what modules are public
+and what modules are internal -- and the objects within those modules -- which
+are public and how this is communicated. This APE does not address any other
+questions of API design, e.g. public versus internal methods in classes, leaving
+these for future APEs or future revisions to this APE.
+
+
+.. _description_of_the_problem:
+
+Description of the Problem
+--------------------------
+
+Pure Python has no language-level means to establish and enforce separations
+between public versus internal interfaces -- all code objects can be accessed.
+This is in contrast to languages like C++ and Java, which have language-level
+constructs to establish public versus private interfaces. Differentiating
+between public and internal interfaces is important for everyone, users and
+developers alike, to know what interfaces are stable and supported, and which
+are not. Consequently, Python has developed conventions for communicating what
+is public versus internal. Nominally there are three means (discussed more
+formally later):
+
+1. **Underscore Prefix**.  By convention, names that begin with an underscore
```

Comment:
technicality: name mangling rules are confusing, let's stay away from them if we can (see https://docs.python.org/3/tutorial/classes.html#private-variables)
```suggestion
1. **Underscore Prefix**.  By convention, names that begin with a single underscore
```